### PR TITLE
Refactor/update routing

### PR DESCRIPTION
**Pull request by martynpclark**
Further speed up

### CODE REVIEW

File: route/build/src/network_topo.f90 [hunk 1]
```diff
@@ -717,6 +718,148 @@ SUBROUTINE REACH_LIST(&
  IMPLICIT NONE
  ! input variables
  INTEGER(I4B)      , INTENT(IN)                 :: NRCH            ! number of stream segments
+ logical(lgt)      , intent(in)                 :: doReachList     ! flag to compute the list of upstream reaches
+ type(var_ilength) , intent(inout), allocatable :: structNTOPO(:)  ! network topology structure
+ ! output variables
+ type(var_dlength) , intent(inout)              :: structSEG(:)    ! network topology structure
```
Comment (nmizukami) [inline route/build/src/network_topo.f90]: comment should be "reach property structure" ?

File: route/build/src/network_topo.f90 [hunk 2]
```diff
@@ -701,9 +701,10 @@ END SUBROUTINE REACH_MASK
  SUBROUTINE REACH_LIST(&
                        ! input
                        NRCH,        & ! Number of reaches
-                       structNTOPO, & ! Network topology
                        doReachList, & ! flag to compute the list of upstream reaches
+                       structNTOPO, & ! Network topology
                        ! output
+                       structSEG,   & ! Network properties
```
Comment (nmizukami) [inline route/build/src/network_topo.f90]: comment: "reach properties"?

File: route/build/src/network_topo.f90 [hunk 3]
```diff
@@ -717,6 +718,148 @@ SUBROUTINE REACH_LIST(&
...
+ logical(lgt)      , intent(in)                 :: doReachList     ! flag to compute the list of upstream reaches
+ type(var_ilength) , intent(inout), allocatable :: structNTOPO(:)  ! network topology structure
+ ! output variables
+ type(var_dlength) , intent(inout)              :: structSEG(:)    ! network topology structure
+ integer(i4b)      , intent(out)                :: NTOTAL          ! total number of upstream reaches for all reaches
+ integer(i4b)      , intent(out)                :: ierr            ! error code
+ character(*)      , intent(out)                :: message         ! error message
+ ! local variables
+ INTEGER(I4B)                                   :: IRCH,JRCH,KRCH  ! loop through reaches
+ logical(lgt)                                   :: processedReach(nRch)  ! flag to define if reaches are processed
```
Comment (nmizukami) [inline route/build/src/network_topo.f90]: you prefer perfect alignment on comment?

File: route/build/src/network_topo.f90 (hunk 2 above)
Comment (martynpclark) [reply route/build/src/network_topo.f90]: Fixed

File: route/build/src/network_topo.f90 (hunk 3 above)
Comment (martynpclark) [reply route/build/src/network_topo.f90]: No

File: route/build/src/network_topo.f90 (hunk 1 above)
Comment (martynpclark) [reply route/build/src/network_topo.f90]: Fixed

**Pull request merged**